DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/annotated_ptr/src/annotated_ptr.cpp

```diff
@@ -0,0 +1,191 @@
+#include <iomanip>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+using Pipe2DotProductIP =
+    sycl::ext::intel::experimental::pipe<class MyPipeName1, float *>;
+using Pipe2AnnotatedPtrIP =
+    sycl::ext::intel::experimental::pipe<class MyPipeName2, float *>;
+
+constexpr int kBL1 = 1;
+constexpr int kBL2 = 2;
+
+constexpr int ROWS = 2;
+constexpr int COLS = 5;
+
+// The kernel 'DotProductIP' computes a weighted sum over a matrix and a vector:
+// out_vec[0] = mat[0][0] * in_vec[0] + mat[0][1] * in_vec[1] + ... +
+// out_vec[1] = mat[1][0] * in_vec[0] + mat[1][1] * in_vec[1] + ... +
+//          ...
+struct DotProductIP {
+  sycl::ext::oneapi::experimental::annotated_arg<
+      float *, decltype(sycl::ext::oneapi::experimental::properties{
+                   sycl::ext::intel::experimental::buffer_location<kBL2>})>
+      in_vec;
+  sycl::ext::oneapi::experimental::annotated_arg<
+      float *, decltype(sycl::ext::oneapi::experimental::properties{
+                   sycl::ext::intel::experimental::buffer_location<kBL1>})>
+      out_vec;
+
+  void operator()() const {
+    for (int i = 0; i < ROWS; i++) {
+      // read the starting pointer of the i-th row of the weight matrix
+      float *p = Pipe2DotProductIP::read();
+
+      float sum = 0.0f;
+#pragma unroll COLS
+      for (int j = 0; j < COLS; j++) sum += p[j] * in_vec[j];
+
+      out_vec[i] = sum;
+    }
+  }
+};
+
+// The kernel 'AnnotatedPtrIP' computes the same function, but with
+// `annotated_ptr` specifying the buffer location of pointers read from the host
+// pipe.
+struct AnnotatedPtrIP {
+  sycl::ext::oneapi::experimental::annotated_arg<
+      float *, decltype(sycl::ext::oneapi::experimental::properties{
+                   sycl::ext::intel::experimental::buffer_location<kBL2>})>
+      in_vec;
+  sycl::ext::oneapi::experimental::annotated_arg<
+      float *, decltype(sycl::ext::oneapi::experimental::properties{
+                   sycl::ext::intel::experimental::buffer_location<kBL1>})>
+      out_vec;
+
+  void operator()() const {
+    for (int i = 0; i < ROWS; i++) {
+      // read the starting pointer of the i-th row of the weight matrix
+      float *p = Pipe2AnnotatedPtrIP::read();
+
+      // set buffer location on p with annotated_ptr
+      sycl::ext::oneapi::experimental::annotated_ptr<
+          float, decltype(sycl::ext::oneapi::experimental::properties{
+                     sycl::ext::intel::experimental::buffer_location<kBL1>})>
+          mat{p};
+
+      float sum = 0.0f;
+#pragma unroll COLS
+      for (int j = 0; j < COLS; j++) sum += mat[j] * in_vec[j];
+
+      out_vec[i] = sum;
+    }
+  }
+};
+
+// verify results
+bool check_result(float *result, float *expected, int size) {
```

Comment:
```suggestion
bool CheckResult(float *result, float *expected, int size) {
```

Function names need to be `PascalCase`